### ISSUE TITLE
[Error] CUDA Too many resources requested for launch

### ISSUE THREAD

**Issue by Tachimura** [junzhezhang/mesh-inversion#4]
Hello, i'm trying to run your work but i'm getting an error when i launch the run_inversion.py script.

# Setup
I'm working on a system with:
- 64GB RAM
- 2 RTX 2080ti (12GB VRAM each)
- ubuntu 20.04

# Installation procedure i've followed
### Kaolin Setup

1. Download kaolin (commit [e7e5131](https://github.com/NVIDIAGameWorks/kaolin/tree/e7e513173bd4159ae45be6b3e156a3ad156a3eb9))
2. Create environment with: "conda create --name mesh-inversion python=3.7" and then activate it via "conda activate mesh-inversion"
3. Build kaolin following the steps defined in their git page:
	- `conda install pytorch==1.11.0 torchvision==0.12.0 cudatoolkit==11.3.1 -c python`
	- `cd kaolin/`
	- `python setup.py build_ext --inplace`
	- `python setup.py install`
	- `pip install --no-dependencies nuscenes-devkit opencv-python-headless scikit-learn joblib pyquaternion cachetools`
	- [test kaolin] => `python -c "import kaolin; print(kaolin.__version__)"` which prints 0.1.0 after installing the missing lib 'threadpoolctl' via pip

### Mesh-Inversion setup
1. Install some requisite packages for mesh_inversion:
	- `pip install absl-py packaging tensorboard`
2. In `run_inversion.py` script i've added a print in the for loop just to show the current loop status `print(f"Image: {idx}")`
	(I get the same error even without this print)

### Code Execution
Now i run the mesh_inversion.py script:
`python mesh_inversion.py --name author_released --checkpoint_dir pretrained`

# Error
![error](https://user-images.githubusercontent.com/62809228/203556150-d6a986eb-f948-4228-8463-825708b30bad.jpg)
It takes about 20-30 seconds for the error to appear after the "Image 0" print (which is done in row 38 of run_inversion.py right after the code:
 `idx = data['idx'][0].item()`)

# Successive Attempts
- I've tried using "--num_workers 0" and "--num_workers 1"	-> i still get the same error.
- I've tried using "export CUDA_VISIBLE_DEVICES=0,1"	-> i still get the same error.
- I've tried using"--gpu_ids 0,1" with and without CUDA_VISIBLE_DEVICES=0,1	-> i still get the same error.
- I've tried using different cuda versions	-> i still get the same error.
- I've also tried with python 3.6	-> i still get the same error.

Do you have any idea of what is going wrong?

Thanks in advance for your support!

**Comment by junzhezhang** [comment]
Hi,

Perhaps it is due to higher resources required than your CUDA can provide. You may want to explore reducing the sampling points of texture loss from 8192 to say 1024. Besides, you may also want to verify with a lower version of cuda, since I never tested on 2080Ti. 

Hope it helps.

Junzhe

**Comment by mitesuti** [comment]
Sorry, I have the same problem as you. 
How can I reduce the sampling points for texture loss?
Thank you in advance for your help.
